vir/src/data.rs

```diff
@@ -98,6 +99,13 @@ pub enum TypeData<'vir> {
     // TODO: separate `TyParam` variant? `Domain` used for now
     Ref, // TODO: typed references ?
     Perm,
+    Unsupported(UnsupportedType<'vir>)
+}
+
+#[derive(Clone)]
+pub struct UnsupportedType<'vir> {
+    pub name: &'vir str,
+    pub fallback: Option<Type<'vir>>
```

Comment:
When would the `fallback` be used? Add a comment to the field.

Reply:
Actually, I'm not sure if it would be, so I just removed it for now.